runtime/hydradx/src/evm/erc20_currency.rs

```diff
@@ -259,15 +274,35 @@ where
 		amount: Self::Balance,
 	) -> sp_runtime::DispatchResult {
 		let sender = <pallet_evm_accounts::Pallet<T>>::evm_address(from);
-		<Self as ERC20>::transfer(
-			CallContext {
-				contract,
-				sender,
-				origin: sender,
-			},
-			EvmAccounts::<T>::evm_address(to),
-			amount,
-		)
+		let call_context = CallContext {
+			contract,
+			sender,
+			origin: sender,
+		};
+
+		if let Some(underlying_asset) = AaveTradeExecutor::<T>::get_underlying_asset_from(contract) {
+			let asset_id = HydraErc20Mapping::address_to_asset(contract)
+				.ok_or(DispatchError::Other("No substrate asset found for address"))?;
+			let ed = pallet_asset_registry::Pallet::<T>::existential_deposit(asset_id).unwrap_or_default();
+			let atoken_balance = Self::free_balance(contract, from);
+
+			let diff = atoken_balance.saturating_sub(amount);
+			if diff <= ed {
+				//We withdraw all AToken and supply underlying asset amount on behalf of the receiver
+				//We need to do this as Aave ScaledBalanceTokenBase.sol has rounding in transfer method so we can't always transfer total balance
+				AaveTradeExecutor::<T>::withdraw_all(from, underlying_asset)?;
+				let underlying_balance = Self::free_balance(underlying_asset, from);
```

Comment:
this would supply all tokens you had, not just those you withdrawn! you should only transfer delta of underlying asset

Reply:
fixed, pls double check